### PR TITLE
Closes #50

### LINKED ISSUE #50: Create a dataset loader for SciCite
From https://github.com/allenai/scicite

### PR DESCRIPTION
**Pull request by stephenbach**
- **Name:** SciCite
- **Description:** SciCite is a dataset of 11K manually annotated citation intents based on citation context in the computer science and biomedical domains.
- **Paper:** https://aclanthology.org/N19-1361/
- **Data:** https://allenai.org/data/scicite

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by stephenbach** [comment]
Note for reviewers: this dataset has a bunch of missing values. Per [this slack thread](https://huggingface.slack.com/archives/C020T0W9PBK/p1649970184493709), I am not using BigBioValues due to some errors. I have followed the [implementation in HF datasets](https://github.com/huggingface/datasets/blob/master/datasets/scicite/scicite.py) instead. There is a TODO in the doc string.

File: biodatasets/scicite/scicite.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+import datasets
+import numpy as np
+
+from utils import schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+_CITATION = """\
+@inproceedings{,
```
Comment (stephenbach) [inline biodatasets/scicite/scicite.py:43]:
```suggestion
@inproceedings{cohan:naacl19,
```

**Review by ruisi-su — APPROVED** [review]
LGTM! Thank you for your contribution @stephenbach 🙌

File: biodatasets/scicite/scicite.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+_DATASETNAME = "scicite"
+
+_DESCRIPTION = """\
+SciCite is a dataset of 11K manually annotated citation intents based on
+citation context in the computer science and biomedical domains.
+"""
+
+_HOMEPAGE = "https://allenai.org/data/scicite"
+
+_LICENSE = ""
```
Comment (ruisi-su) [inline biodatasets/scicite/scicite.py:62]: @stephenbach I checked the scicite repo, and it says the data is under Apache License 2.0. Would it be possible to have it here?

Comment (stephenbach) [reply biodatasets/scicite/scicite.py:62]:
```suggestion
_LICENSE = "Apache License 2.0"
```
Hi @ruisi-su IANAL, but I read that to mean that the code in the repo is Apache 2.0, but there's no license on the data itself, which is just linked from the README. Left the change as a suggestion if you want to commit it. Thanks!

Comment (jason-fries) [reply biodatasets/scicite/scicite.py:62]: Thanks @stephenbach and @ruisi-su ! Let's leave the `_LICENSE` as empty for this merge due to the ambiguity.

**Pull request merged**